File: Projeto_6/monitor.cpp
#include "Monitor.h"
#include <iostream>

Monitor::Monitor(const std::string& nome, const std::string& email, const std::string& matricula, const std::string& curso, const std::string& areaDeAtuacao, const std::string& senha)
    : UsuarioAutenticavel(nome, email, "MONITOR", senha) {}

void Monitor::adicionarDisciplinaMonitorada(const std::string& disciplina) {
    disciplinasMonitoradas.push_back(disciplina);
}

void Monitor::gerarRelatorio() const {
    std::cout << "Relatório do Monitor\n";
    std::cout << "Nome: " << nome << "\nEmail: " << email << "\n";
    std::cout << "Disciplinas Monitoradas: ";
    for (const auto& disc : disciplinasMonitoradas) {
        std::cout << disc << " ";
    }
    std::cout << "\n";
}

File: Projeto_6/src/monitor.py
from aluno import Aluno
from professor import Professor

class Monitor(Aluno, Professor):
    def __init__(self, nome, email, matricula, curso, area_de_atuacao, disciplinas_ministradas=None):
        # Inicializa atributos de Aluno e Professor
        Aluno.__init__(self, nome, email, matricula, curso)
        Professor.__init__(self, nome, email, area_de_atuacao, disciplinas_ministradas)

    def listar_disciplinas_monitoradas(self):
        # Lista as disciplinas monitoradas pelo monitor
        print(f"Monitor: {self.nome}")
        print(f"Disciplinas Monitoradas: ", ", ".join(self.disciplinas_ministradas))
